source/AndroidResolver/src/PlayServicesResolver.cs

```diff
@@ -2191,15 +2192,56 @@ public static void SettingsDialog()
                 // https://docs.gradle.org/3.4/release-notes.html#the-java-library-plugin
                 // https://developer.android.com/studio/releases/gradle-plugin#3-0-0
                 var version = GradleVersion;
+                var versionComparer = new Dependency.VersionComparer();
                 var includeStatement =
                     !String.IsNullOrEmpty(version) &&
-                    (new Dependency.VersionComparer()).Compare("3.4", version) >= 0 ?
+                    versionComparer.Compare("3.4", version) >= 0 ?
                     "implementation" : "compile";
                 if (includeDependenciesBlock) lines.Add("dependencies {");
+                // Build a map of dependencies with the max version of that dependency.
+                // If different packages have different versions of the same dependency,
+                // we want to activate only the highest version but still include the
+                // other versions as commented out dependency lines.
+                var dependenciesMaxVersions = new Dictionary<string, Dependency>();
+                Dependency dependencyMaxVersion;
```

Comment:
This variable can be in the first level of the loop below.  Otherwise, it looks like this is a variable used cross different iteration of the loop.

Reply:
Done